### PR TITLE
md2gemini: init at 1.8.1

### PR DESCRIPTION
**Pull request by KAction**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4279]: I think it would be clearer if we converted the current default into a generic nix and created a 0.8.nix and a 2.nix file.

File: pkgs/development/python-modules/md2gemini/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/md2gemini/default.nix:18]: Missing maintainer.

File: pkgs/development/python-modules/md2gemini/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/md2gemini/default.nix:8]:
```suggestion
```

File: pkgs/development/python-modules/md2gemini/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/md2gemini/default.nix:14]:
```suggestion

  propagatedBuildInputs = [ mistune_2 cjkwrap wcwidth ];
  checkInputs = [ pytestCheckHook ];
  pythonImportsCheck = [ "md2gemini" ];

```

pytest alone does not run pytest.

File: pkgs/development/python-modules/md2gemini/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/md2gemini/default.nix:1]:
```suggestion
{ lib, buildPythonPackage, fetchPypi, mistune_2, cjkwrap, wcwidth, pytestCheckHook }:
```

File: pkgs/development/python-modules/cjkwrap/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/cjkwrap/default.nix:14]: Missing maintainer.

File: pkgs/development/python-modules/cjkwrap/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/cjkwrap/default.nix:6]:
```suggestion
  version = "2.2";

  src = fetchPypi {
```

File: pkgs/development/python-modules/cjkwrap/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/cjkwrap/default.nix:10]:
```suggestion

  pythonImportsCheck = [ "cjkwrap" ];

```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:671]: That causes an eval error.

File: pkgs/top-level/python-packages.nix
Comment (KAction) [reply pkgs/top-level/python-packages.nix]: Did it same way as postgresql do.

File: pkgs/development/python-modules/md2gemini/default.nix
Comment (KAction) [reply pkgs/development/python-modules/md2gemini/default.nix:20]: Added.

File: pkgs/top-level/all-packages.nix
Comment (KAction) [reply pkgs/top-level/all-packages.nix]: That does not belong here, removed.

**Comment by KAction** [comment]
Looks like I applied all suggestions.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by KAction** [comment]
Okay, now @ofborg is happy.

File: pkgs/development/python-modules/mistune/common.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mistune/common.nix:15]: Missing maintainer.

File: pkgs/development/python-modules/mistune/common.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mistune/common.nix:11]: Missing checkPhase and pythonImportsCheck.

File: pkgs/development/python-modules/mistune/common.nix
Comment (KAction) [reply pkgs/development/python-modules/mistune/common.nix:15]: It was this way before.

File: pkgs/development/python-modules/mistune/common.nix
Comment (KAction) [reply pkgs/development/python-modules/mistune/common.nix:11]: Same here.

**Comment by KAction** [comment]
Okay, I added pythonImportsCheck to mistune, but I'd rather not assume maintainership of every indirect dependency I touch. Wh

**Comment by SuperSandro2000** [comment]
> Okay, I added pythonImportsCheck to mistune, but I'd rather not assume maintainership of every indirect dependency I touch. Wh

The problem is that if your dependencies are not maintained they are potentially in a worse state but it seems like mistune is used in a lot more packages.

I think adding the imports check to mistune triggered the mass rebuild. We really want to have one if there are not tests because otherwise the package can silently break. I think the best solution would be to merge this PR into staging because if we split the change out we force a merge conflict which someone would need to resolve. Let me know if thats a problem for you.

**Review by SuperSandro2000 — APPROVED** [review]
other than that the ordering could be a bit more standard LGTM if rebased to staging

**Review by SuperSandro2000 — APPROVED** [review]
other than that the ordering could be a bit more standard LGTM if rebased to staging

**Comment by KAction** [comment]
Yes, sure. Let's merge it into staging.

**Pull request merged**
